### PR TITLE
Transaction queue integration + submission RPC

### PR DESCRIPTION
**Pull request by gavofyork**
Goes in after #105.

### CODE REVIEW

File: substrate/client/src/client.rs [hunk 1]
```diff
@@ -136,6 +136,13 @@ impl<B, E> Client<B, E> where
 		})
 	}
 
+	/// Submit a transaction to this for potential inclusion in a block.
+	pub fn submit_transaction(&self, _tx: block::Transaction) -> error::Result<()> {
+		// TODO: introduce a tx queue and place the transaction in there. tx queue or some of its
```
Comment (rphmeier) [inline substrate/client/src/client.rs]: 1. I don't think this belongs in client
2. substrate blocks don't necessarily have transactions, so this is something that belongs in chain-specific code (tx ordering is completely chain dependent)

File: substrate/rpc/src/author/mod.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+#[cfg(test)]
+mod tests;
+
+use self::error::{Result};
+
+build_rpc_trait! {
+	/// Polkadot blockchain API
+	pub trait AuthorApi {
+		/// Get header of a relay chain block.
```
Comment (rphmeier) [inline substrate/rpc/src/author/mod.rs]: substrate doesn't have a notion of a relay chain

File: substrate/client/src/client.rs (hunk 1 above)
Comment (arkpar) [reply substrate/client/src/client.rs]: Can be removed now.

File: polkadot/transaction-pool/src/lib.rs [hunk 3]
```diff
@@ -371,6 +374,16 @@ impl TransactionPool {
 	}
 }
 
+impl substrate_rpc::author::AsyncAuthorApi for TransactionPool {
+	fn submit_extrinsic(&mut self, xt: Extrinsic) -> substrate_rpc::author::error::Result<()> {
+		self.import(xt
+			.using_encoded(|ref mut s| UncheckedExtrinsic::decode(s))
+			.ok_or(substrate_rpc::author::error::ErrorKind::InvalidFormat)?
+		).map(|_| ())
```
Comment (rphmeier) [inline polkadot/transaction-pool/src/lib.rs]: `map` should be on following line

File: substrate/client/src/client.rs (hunk 1 above)
Comment (rphmeier) [reply substrate/client/src/client.rs]: @gavofyork this is dead code now, we should remove it

**Pull request merged**